src/System.Private.CoreLib/shared/System/Collections/ObjectModel/Collection.cs

```diff
@@ -116,7 +116,25 @@ public void Insert(int index, T item)
             InsertItem(index, item);
         }
 
-        public void InsertRange(int index, IEnumerable<T> collection) => InsertItemsRange(index, collection);
+        public void InsertRange(int index, IEnumerable<T> collection)
+        {
+            if (items.IsReadOnly)
+            {
+                ThrowHelper.ThrowNotSupportedException(ExceptionResource.NotSupported_ReadOnlyCollection);
+            }
+
+            if (collection == null)
+            {
+                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.list);
```

Comment:
There's no `list` argument to this method, but the exception is going to be `new ArgumentNullException("list")`.  The `ExceptionArgument.list` should be changed to `ExceptionArgument.collection`.